\documentclass[10pt,letterpaper]{article}

\usepackage{eso-pic}

\setlength\parindent{0pt}

\AddToShipoutPictureBG{%

\ifnum\value{page}>1{
\AtTextUpperLeft{
\makebox[18.5cm][r]{
\raisebox{-2.3cm}{%
{\transparent{0.3}{\includegraphics[width=0.29\textwidth]{e-logo.png}}	}} } }
}\fi
}

\AddToShipoutPicture{%
{
 {\color{blGreen!70!red}\transparent{0.9}{\put(0,0){\rule{3pt}{\paperheight}}}}%
 {\color{darkRed!70!purple}\transparent{1}\put(3,0){{\rule{4pt}{\paperheight}}}}
% {\color{logoPeach!80!cyan}\transparent{0.5}{\put(0,700){\rule{1cm}{.6cm}}}}%
% {\color{darkRed!60!cyan}\transparent{0.7}\put(0,706){{\rule{1cm}{.6cm}}}}
% \put(18,726){\thepage}
% \transparent{0.8}
}
}



\AddToShipoutPicture{%

\ifnum\value{page}>0


{\color{blGreen!70!red}\transparent{0.9}{\put(300,8){\rule{0.5\paperwidth}{.3cm}}}}%
{\color{inOne}\transparent{0.8}{\put(300,10){\rule{0.5\paperwidth}{.3cm}}}}%
{\color{inTwo}\transparent{0.3}\put(300,13){{\rule{0.5\paperwidth}{.3cm}}}}

\put(301,16){%
\transparent{0.7}{
\includegraphics[width=0.2\textwidth]{logo.png}} }

{\color{blGreen!70!red}\transparent{0.9}{\put(5.6,5){\rule{0.5\paperwidth}{.4cm}}}}%
{\color{inOne}\transparent{1}{\put(5.6,10){\rule{0.5\paperwidth}{.4cm}}}}%
{\color{inTwo}\transparent{0.3}\put(5.6,15){{\rule{0.5\paperwidth}{.4cm}}}}

\fi
}

%\pagestyle{empty} % no page number
%\parskip 7.2pt    % space between paragraphs
%\parindent 12pt   % indent for new paragraph
%\textwidth 4.5in  % width of text
%\columnsep 0.8in  % separation between columns

\setlength{\footskip}{23pt}

\usepackage[paperheight=17.25in,paperwidth=9.1in]{geometry}
\geometry{left=1.3in,top=1.1in,right=.6in,bottom=1.6in} %margins

\newcommand{\sectsp}{\vspace{12pt}}

\usepackage{graphicx}
\usepackage{color,framed}

\usepackage{float}

\usepackage{mdframed}


\usepackage{setspace}
\newcommand{\rpdfNotice}[1]{\begin{onehalfspacing}{

\Large #1

}\end{onehalfspacing}}

\usepackage{xcolor}

\usepackage[hyphenbreaks]{breakurl}
\usepackage[hyphens]{url}

\usepackage{hyperref}
\newcommand{\rpdfLink}[1]{\href{#1}{\small{#1}}}
\newcommand{\dblHref}[1]{\href{#1}{\small{\burl{#1}}}}
\newcommand{\browseHref}[2]{\href{#1}{\Large #2}}

\colorlet{blCyan}{cyan!50!blue}

\definecolor{darkRed}{rgb}{.2,.0,.1}

\hypersetup{
    colorlinks=true,
    linkcolor=blCyan!80!darkRed,
    filecolor=magenta,
    urlcolor=blue,
}

\urlstyle{same}

\definecolor{blGreen}{rgb}{.2,.7,.3}

\definecolor{darkBlGreen}{rgb}{.1,.3,.2}

\definecolor{oldBlColor}{rgb}{.2,.7,.3}

\definecolor{blColor}{rgb}{.1,.3,.2}

\definecolor{elColor}{rgb}{.2,.1,0}
\definecolor{flColor}{rgb}{0.7,0.3,0.3}

\definecolor{logoOrange}{RGB}{108, 18, 30}
\definecolor{logoGreen}{RGB}{85, 153, 89}
\definecolor{logoPurple}{RGB}{200, 208, 30}

\definecolor{logoBlue}{RGB}{4, 2, 25}
\definecolor{logoPeach}{RGB}{255, 159, 102}
\definecolor{logoCyan}{RGB}{66, 206, 244}
\definecolor{logoRed}{rgb}{.3,0,0}

\definecolor{inOne}{rgb}{0.122, 0.435, 0.698}% Rule colour
\definecolor{inTwo}{rgb}{0.122, 0.698, 0.435}% Rule colour

\definecolor{outOne}{rgb}{0.435, 0.698, 0.122}% Rule colour
\definecolor{outTwo}{rgb}{0.698, 0.435, 0.122}% Rule colour

\usepackage[many]{tcolorbox}% http://ctan.org/pkg/tcolorbox

\usepackage{transparent}

\newenvironment{cframed}{\begin{mdframed}[linecolor=logoPeach,linewidth=0.4mm]}{\end{mdframed}}

\newenvironment{ccframed}{\begin{mdframed}[backgroundcolor=logoGreen!5,linecolor=logoCyan!50!black,linewidth=0.4mm]}{\end{mdframed}}

\usepackage{aurical}
\usepackage[T1]{fontenc}

\usepackage{relsize}

\newcommand{\bref}[1]{\hspace*{1pt}\textbf{\ref{#1}}}

\newcommand{\pseudoIndent}{

\vspace{10pt}\hspace*{12pt}}

\newcommand{\YPDFI}{{\fontfamily{fvs}\selectfont YPDF-Interactive}}

%
\newcommand{\deconum}[1]{{\protect\raisebox{-1pt}{{\LARGE #1}}}}

\newcommand{\visavis}{vis-\`a-vis}

\newcommand{\VersatileUX}{{\color{red!85!black}{\Fontauri Versatile}}%
{{\fontfamily{qhv}\selectfont\smaller UX}}}

\newcommand{\NDPCloud}{{\color{red!15!black}%
{\fontfamily{qhv}\selectfont {\smaller NDP C{\smaller LOUD}}}}}

\newcommand{\lfNDPCloud}{{\color{red!15!black}%
{\fontfamily{qhv}\selectfont N{\smaller DP C{\smaller LOUD}}}}}

\newcommand{\textds}[1]{{\fontfamily{lmdh}\selectfont{%
\raisebox{-1pt}{#1}}}}

\newcommand{\dsC}{{\textds{ds}{\fontfamily{qhv}\selectfont \raisebox{-1pt}
{\color{red!15!black}{C}}}}}

\definecolor{tcolor}{RGB}{24,52,61}

\newcommand{\CCpp}{\resizebox{!}{8pt}{\AcronymText{C}/\AcronymText{C++}}}
\newcommand{\NoSQL}{\resizebox{!}{8pt}{\AcronymText{NoSQL}}}
\newcommand{\SQL}{\resizebox{!}{8pt}{\AcronymText{SQL}}}

\newcommand{\NCBI}{\resizebox{!}{8pt}{\AcronymText{NCBI}}}

\newcommand{\HTXN}{\resizebox{!}{8pt}{\AcronymText{HTXN}}}
\newcommand{\lHTXN}{\resizebox{!}{8.5pt}{\AcronymText{HTXN}}}
\newcommand{\lsHTXN}{\resizebox{!}{9.5pt}{\AcronymText{\textcolor{tcolor}{HTXN}}}}


\usepackage{mdframed}

\newcommand{\cframedboxpanda}[1]{\begin{mdframed}[linecolor=yellow!70!blue,linewidth=0.4mm]#1\end{mdframed}}


\newcommand{\PVD}{\resizebox{!}{8pt}{\AcronymText{PVD}}}

\newcommand{\sapp}{\resizebox{!}{8pt}{\AcronymText{Sapien+}}}
\newcommand{\lsapp}{\resizebox{!}{8.5pt}{\AcronymText{Sapien+}}}
\newcommand{\lssapp}{\resizebox{!}{9.5pt}{\AcronymText{Sapien+}}}

\newcommand{\ePub}{\resizebox{!}{8pt}{\AcronymText{ePub}}}

%\lsLPF


\newcommand{\GIT}{\resizebox{!}{8pt}{\AcronymText{GIT}}}

\newcommand{\LPF}{\resizebox{!}{8pt}{\AcronymText{LPF}}}
\newcommand{\lLPF}{\resizebox{!}{8.5pt}{\AcronymText{LPF}}}
\newcommand{\lsLPF}{\resizebox{!}{9.5pt}{\AcronymText{LPF}}}

\makeatletter

\newcommand*\getX[1]{\expandafter\getX@i#1\@nil}

\newcommand*\getY[1]{\expandafter\getY@i#1\@nil}
\def\getX@i#1,#2\@nil{#1}
\def\getY@i#1,#2\@nil{#2}
\makeatother
	
\newcommand{\rectann}[9]{%
\path [draw=#1,draw opacity=#2,line width=#3, fill=#4, fill opacity = #5, even odd rule] %
(#6) rectangle(\getX{#6}+#7,\getY{#6}+#8)
({\getX{#6}+((#7-(#7*#9))/2)},{\getY{#6}+((#8-(#8*#9))/2)}) rectangle %
({\getX{#6}+((#7-(#7*#9))/2)+#7*#9},{\getY{#6}+((#8-(#8*#9))/2)+#8*#9});}


\definecolor{pfcolor}{RGB}{94, 54, 73}

\newcommand{\EPF}{\resizebox{!}{8pt}{\AcronymText{ETS{\color{pfcolor}pf}}}}
\newcommand{\lEPF}{\resizebox{!}{8.5pt}{\AcronymText{ETS{\color{pfcolor}pf}}}}
\newcommand{\lsEPF}{\resizebox{!}{9.5pt}{\AcronymText{ETS{\color{pfcolor}pf}}}}


\newcommand{\XPDF}{\resizebox{!}{8.5pt}{\AcronymText{XPDF}}}

\newcommand{\GRE}{\resizebox{!}{8.5pt}{\AcronymText{GRE}}}

\newcommand{\lMOSAIC}{\resizebox{!}{8.5pt}{\AcronymText{MOSAIC}}}

\newcommand{\XML}{\resizebox{!}{8pt}{\AcronymText{XML}}}
\newcommand{\RDF}{\resizebox{!}{8pt}{\AcronymText{RDF}}}
\newcommand{\DOM}{\resizebox{!}{8pt}{\AcronymText{DOM}}}

\newcommand{\CLang}{\resizebox{!}{8pt}{\AcronymText{C}}}

\newcommand{\HNaN}{\resizebox{!}{8pt}{\AcronymText{HN%
\textsc{a}N}}}

\newcommand{\JSON}{\resizebox{!}{8pt}{\AcronymText{JSON}}}

\newcommand{\MeshLab}{\resizebox{!}{8pt}{\AcronymText{MeshLab}}}
\newcommand{\IQmol}{\resizebox{!}{8pt}{\AcronymText{IQmol}}}

\newcommand{\SGML}{\resizebox{!}{8pt}{\AcronymText{SGML}}}

\newcommand{\GUI}{\resizebox{!}{8pt}{\AcronymText{GUI}}}

\newcommand{\API}{\resizebox{!}{8pt}{\AcronymText{API}}}

\newcommand{\SDI}{\resizebox{!}{8pt}{\AcronymText{SDI}}}

\newcommand{\IDE}{\resizebox{!}{8pt}{\AcronymText{IDE}}}

\newcommand{\ThreeD}{\resizebox{!}{8pt}{\AcronymText{3D}}}

\newcommand{\FAIR}{\resizebox{!}{8pt}{\AcronymText{FAIR}}}

\newcommand{\QNetworkManager}{\resizebox{!}{8pt}{\AcronymText{QNetworkManager}}}
\newcommand{\QTextDocument}{\resizebox{!}{8pt}{\AcronymText{QTextDocument}}}
\newcommand{\QWebEngineView}{\resizebox{!}{8pt}{\AcronymText{QWebEngineView}}}
\newcommand{\HTTP}{\resizebox{!}{8pt}{\AcronymText{HTTP}}}


\newcommand{\lAcronymTextNC}[2]{{\fontfamily{fvs}\selectfont {\Large{#1}}{\large{#2}}}}

\newcommand{\AcronymTextNC}[1]{{\fontfamily{fvs}\selectfont {\large #1}}}


\colorlet{orr}{orange!60!red}

\newcommand{\textscc}[1]{{\color{orr!35!black}{{%
						\fontfamily{Cabin-TLF}\fontseries{b}\selectfont{\textsc{\scriptsize{#1}}}}}}}


\newcommand{\textsccserif}[1]{{\color{orr!35!black}{{%
				\scriptsize{\textbf{#1}}}}}}


\newcommand{\iXPDF}{\resizebox{!}{8pt}{\textsccserif{%
\textit{XPDF}}}}

\newcommand{\iEPF}{\resizebox{!}{8pt}{\textsccserif{%
\textit{ETSpf}}}}

\newcommand{\iSDI}{\resizebox{!}{8pt}{\textsccserif{%
\textit{SDI}}}}

\newcommand{\iHTXN}{\resizebox{!}{8pt}{\textsccserif{%
\textit{HTXN}}}}


\newcommand{\AcronymText}[1]{{\textscc{#1}}}

\newcommand{\AcronymTextser}[1]{{\textsccserif{#1}}}


\newcommand{\mAcronymText}[1]{{\textscc{\normalsize{#1}}}}

\newcommand{\FASTA}{{\resizebox{!}{8pt}{\AcronymText{FASTA}}}}
\newcommand{\SRA}{{\resizebox{!}{8pt}{\AcronymText{SRA}}}}
\newcommand{\DNA}{{\resizebox{!}{8pt}{\AcronymText{DNA}}}}
\newcommand{\MAP}{{\resizebox{!}{8pt}{\AcronymText{MAP}}}}
\newcommand{\EPS}{{\resizebox{!}{8pt}{\AcronymText{EPS}}}}
\newcommand{\CSV}{{\resizebox{!}{8pt}{\AcronymText{CSV}}}}
\newcommand{\PDB}{{\resizebox{!}{8pt}{\AcronymText{PDB}}}}

\newcommand{\TeXMECS}{\resizebox{!}{8pt}{\AcronymText{TeXMECS}}}

\newcommand{\NGML}{\resizebox{!}{8pt}{\AcronymText{NGML}}}

\newcommand{\Cpp}{\resizebox{!}{8.5pt}{\AcronymText{C++}}}

\newcommand{\WhiteDB}{\resizebox{!}{8pt}{\AcronymText{WhiteDB}}}

\colorlet{drp}{darkRed!70!purple}

%\newcommand{\MOSAIC}{{\color{drp}{\AcronymTextNC{\scriptsize{MOSAIC}}}}}

\newcommand{\MOSAIC}{\resizebox{!}{8pt}{\AcronymText{MOSAIC}}}


\newcommand{\mMOSAIC}{{\color{drp}{\AcronymTextNC{\normalsize{MOSAIC}}}}}

\newcommand{\MOSAICVM}{\mMOSAIC-\mAcronymText{VM}}

\newcommand{\sMOSAICVM}{\resizebox{!}{8pt}{\MOSAICVM}}
\newcommand{\sMOSAIC}{\resizebox{!}{8pt}{\MOSAIC}}

\newcommand{\LDOM}{\resizebox{!}{8pt}{\AcronymText{LDOM}}}

\newcommand{\LXCR}{\resizebox{!}{8pt}{\AcronymText{LXCR}}}
\newcommand{\lLXCR}{\resizebox{!}{8.5pt}{\AcronymText{LXCR}}}
\newcommand{\lsLXCR}{\resizebox{!}{9.5pt}{\AcronymText{LXCR}}}

%\newcommand{\lMOSAIC}{{\color{drp}{\lAcronymTextNC{M}{OSAIC}}}}
\newcommand{\lfMOSAIC}{\resizebox{!}{9pt}{{\color{drp}{\lAcronymTextNC{M}{OSAIC}}}}}

\newcommand{\Mosaic}{\resizebox{!}{8pt}{\MOSAIC}}
\newcommand{\MosaicPortal}{{\color{drp}{\AcronymTextNC{MOSAIC Portal}}}}

\newcommand{\RnD}{\resizebox{!}{7.5pt}{\AcronymText{R\&D}}}
\newcommand{\QtCpp}{\resizebox{!}{8.5pt}{\AcronymText{Qt/C++}}}
\newcommand{\Qt}{\resizebox{!}{9pt}{\AcronymText{Qt}}}
\newcommand{\QtSQL}{\resizebox{!}{8pt}{\AcronymText{QtSQL}}}

\newcommand{\HTML}{\resizebox{!}{8pt}{\AcronymText{HTML}}}
\newcommand{\PDF}{\resizebox{!}{8pt}{\AcronymText{PDF}}}

\newcommand{\lGRE}{\resizebox{!}{8pt}{\AcronymText{GRE}}}

\newcommand{\p}[1]{

\vspace{.85em}#1}

\newcommand{\q}[1]{{\fontfamily{qcr}\selectfont ``}#1{\fontfamily{qcr}\selectfont ''}} 

%\newcommand{\deconum}[1]{{\textcircled{#1}}}


\renewcommand{\thesection}{\protect\mbox{\deconum{\Roman{section}}}}
\renewcommand{\thesubsection}{\arabic{section}.\arabic{subsection}}

\newcommand{\llMOSAIC}{\mbox{{\LARGE MOSAIC}}}
%\newcommand{\lfMOSAIC}{\mbox{M\small{OSAIC}}}

\newcommand{\llMosaic}{\llMOSAIC}
\newcommand{\lMosaic}{\lMOSAIC}
\newcommand{\lfMosaic}{\lfMOSAIC}


\newcommand{\llWC}{\mbox{{\LARGE WhiteCharmDB}}}

\newcommand{\llwh}{\mbox{{\LARGE White}}}
\newcommand{\llch}{\mbox{{\LARGE CharmDB}}}

\usepackage{enumitem}

\setlist[description]{%
  topsep=30pt,               % space before start / after end of list
  itemsep=5pt,               % space between items
  font={\bfseries\sffamily}, % set the label font
%  font={\bfseries\sffamily\color{red}}, % if colour is needed
}

\setlist[enumerate]{%
  topsep=3pt,               % space before start / after end of list
  itemsep=-2pt,               % space between items
  font={\bfseries\sffamily}, % set the label font
%  font={\bfseries\sffamily\color{red}}, % if colour is needed
}

%\usepackage{tcolorbox}

\newcommand{\slead}[1]{%
\noindent{\raisebox{2pt}{\relscale{1.15}{{{%
\fcolorbox{logoCyan!50!black}{logoGreen!5}{#1}
}}}}}\hspace{.5em}}


\let\OldLaTeX\LaTeX

\renewcommand{\LaTeX}{\resizebox{!}{8pt}{\color{orr!35!black}{\OldLaTeX}}}

\let\OldTeX\TeX

\renewcommand{\TeX}{\resizebox{!}{8pt}{\color{orr!35!black}{\OldTeX}}}


\newcommand{\LargeLaTeX}{\resizebox{!}{8.5pt}{\color{orr!35!black}{\OldLaTeX}}}


%\setlength\parindent{24pt}
%\input{commands}


\newcommand{\lun}[1]{\raisebox{-4pt}{\fontfamily{qcr}\selectfont{%
\LARGE{\textbf{\textcolor{tcolor}{#1}}}}}\vspace{-2pt}}

\newcommand{\inditem}{\itemindent10pt\item}

\usepackage{soul}

\definecolor{hlcolor}{RGB}{114, 54, 203}
\colorlet{hlcol}{hlcolor!35}
\sethlcolor{hlcol}

\makeatletter
\def\SOUL@hlpreamble{%
	\setul{}{3ex}%         !!!change this value!!! default is 2.5ex
	\let\SOUL@stcolor\SOUL@hlcolor
	\SOUL@stpreamble
}
\makeatother

\usepackage{scrextend}
%\vspace*{3em}
\newenvironment{mldescription}{\vspace{1em}%
  \begin{addmargin}[4pt]{1em}
    \setlength{\parindent}{-1em}%
    \newcommand*{\mlitem}[1][]{\vspace{5pt}\par\medskip%
%\colorbox{hlcolor}{\textbf{##1}}\quad}\indent
\hl{ \textbf{##1} }\quad}\indent
}{%
  \end{addmargin}
  \medskip
}

\usepackage{marginnote}

\newcommand{\mnote}[1]{%
\vspace*{-2em}
\reversemarginpar
\raisebox{1em}{\marginnote{\parbox{4em}{%
\begin{mdframed}[innerleftmargin=4pt,
	innerrightmargin=1pt,innertopmargin=1pt,
	linecolor=red!20!cyan,userdefinedwidth=4em,
	topline=false,
	rightline=false]
{{\fontfamily{ppl}\fontsize{12}{0}\selectfont
		\textit{#1}}}
\end{mdframed}}
	}[3em]}}

\newcommand{\mnotel}[1]{%
\vspace*{-2em}
\reversemarginpar
\raisebox{-4em}{\marginnote{\parbox{4em}{%
\begin{mdframed}[innerleftmargin=4pt,
	innerrightmargin=1pt,innertopmargin=1pt,
	linecolor=red!20!cyan,userdefinedwidth=4em,
	topline=false,
	rightline=false]
{{\fontfamily{ppl}\fontsize{12}{0}\selectfont
		\textit{#1}}}
\end{mdframed}}
	}[3em]}}

\newcommand{\mnoteh}[3]{%
	\vspace*{#1}
	\reversemarginpar
	\raisebox{#2}{\marginnote{\parbox{4em}{%
				\begin{mdframed}[innerleftmargin=4pt,
					innerrightmargin=1pt,innertopmargin=1pt,
					linecolor=red!20!cyan,userdefinedwidth=4em,
					topline=false,
					rightline=false]
					{{\fontfamily{ppl}\fontsize{12}{0}\selectfont
							\textit{#3}}}
				\end{mdframed}}
			}[3em]}}


\newcommand{\mnoteb}[1]{%
	\vspace*{1em}
	\reversemarginpar
	\raisebox{1em}{\marginnote{\parbox{4em}{%
				\begin{mdframed}[innerleftmargin=4pt,
					innerrightmargin=1pt,innertopmargin=1pt,
					linecolor=red!20!cyan,userdefinedwidth=4em,
					topline=false,
					rightline=false]
					{{\fontfamily{ppl}\fontsize{12}{0}\selectfont
							\textit{#1}}}
				\end{mdframed}}
			}[3em]}}
	
\usepackage{wrapfig}

\usetikzlibrary{arrows, decorations.markings}
\usetikzlibrary{shapes.arrows}

\newcommand{\curicon}[2]{%
	\node at (#1,#2) [
	draw=black,
	%minimum width=2ex,
	inner sep=.7pt,
	fill=white,
	single arrow,
	single arrow head extend=3pt,
	single arrow head indent=1.5pt,
	single arrow tip angle=45,
	line join=bevel,
	minimum height=4.6mm,
	rotate=115
	] {};
}

\begin{document}
	
{\linespread{1.1}\selectfont

\vspace*{-7em}

\begin{center}
%{\relscale{1.2}{\fontfamily{qcr}\fontseries{b}\selectfont 
%{\colorbox{black}{\color{blue}{\llWC{} Database Engine \\and 
%\llMOSAIC{} Native Application Toolkit}}}}}

\colorlet{ctmp}{logoPeach!20!gray}
\colorlet{ctmpp}{ctmp!90!yellow}
\colorlet{ctmppp}{ctmpp!50!black}
\colorlet{ctmpppp}{ctmppp!90!logoRed}

\vspace{1em}


%{\colorbox{darkBlGreen!30!darkRed}{%
\begin{tcolorbox}
[
%%enhanced,
%%frame hidden,
%interior hidden
arc=2pt,outer arc=0pt,
enhanced jigsaw,
width=.984\textwidth,
colback=ctmpppp!30,
colframe=logoRed!30!darkRed,
drop shadow=logoPurple!50!darkRed,
%boxsep=0pt,
%left=0pt,
%right=0pt,
%top=2pt,
]
\begin{minipage}{\textwidth}	
\begin{center}		
{\setlength{\fboxsep}{19pt}
	\relscale{1.4}{{\fontfamily{qcr}\fontseries{b}\selectfont%
{???}}}}
\end{center}
\end{minipage}
\end{tcolorbox}
\end{center}

\vspace{-1.5em}

%\noindent\lun{Overview}
%\fontfamily{ptm}\fontsize{13pt}{18pt}\selectfont
%{\sectsp}
...
\p{As a supplement to the main book, 
several data sets will be published (or 
republished) relevant to subjests covered by 
chapters in the text, such as signal-processing, 
bioacoustics, and Natural Language Processing.  
These datasets' primary purpose is to demonstrate 
data-management and software-development techniques 
discussed in the volume, particularly in the 
third chapter (the data sets are being curated by that 
chapter's author).  These data sets introduce a new 
protocol for publishing research data, which essentially 
extends existing initiatives such as Research Objects 
and \FAIR{} (which stands for \q{Findable, Accessible, 
Interoperable, Reusable}.  In principle, these initiatives 
--- which have been pushed by publishers, academic 
instutitions, and certain government agencies 
(such as the NIH) --- are intended to make 
it easier for scientists to find, assess, 
reuse, and re-produce research data.  
In practice, however --- due partly to  
technological limitations and partly to 
authors and publishers being slow to adopt 
the new protocols --- the ecosystem of 
published data sets remains far less rigorously 
organized than the ecosystem of published 
scientific/academic books and articles.}

\p{The sudden emergence of Covid-19 as a 
medical and governmental priority presents a 
unique case-study of the limitations of our 
existing research-data platforms.  Publishers 
have, to some degree, recognized the extra-ordinary 
nature of the new coronavirus crisis and 
taken some commensurate mearures; for example, 
Springer Nature has committed to releasing as 
open-access documents a collection of papers 
potentially helpful for doctors and 
policymakers responding to the pandemic --- 
some newly published and some dating back several 
years (the earlier research involving viruses 
biologically similar to Covid-19).  As of 
mid-March, this portal encompassed 43 articles, 
of which 15 were accompanied by research data that 
could be separately downloaded (this number does 
not include papers that document research 
findings only indirectly, via tables or graphics 
printed inline with the text).  Collectively 
these articles referenced over 30 distinct data 
sets (some papers were linked to multiple 
data sets), forming a data collection which could 
be a valuable resource for Covid-19 research --- 
not only through the raw data made available 
but as a kernel around which new coronavirus data 
could accumulate.  However, there is currently no 
mechanism to make this overall collection available 
as a single resource.}

\p{This problem demonstrates, among other things, 
how the Research Object protocol is limited in applying 
only to \textit{single} articles.  There is no 
commensurate protocol for publishing \textit{groups} 
of articles which are tied to groups of data sets unified 
into an integral whole.  Open-access Covid-19 papers also 
reveal limitations of exiting online document portals, 
particularly with respect to how publications are 
linked to data sets.  There is no clear indication that a 
given paper is associated with downloaded data; usually 
readers ascertain this information only by reading or 
scrolling down to a \q{supplemental materials} or 
\q{data availability} addendum near the end of the article.  
Moreover, because the Springer Nature portal aggregates 
papers from multiple sources, there is no consistent 
pattern for locating data sets; each journal or 
publisher has their own mechanism for alerting readers 
to the existence of open-access data and allowing them 
to download the relevant data sets.}

\p{Aside from these user-interaction issues, the 
collective group of Covid-19 data sets illustrate 
the limitations of information spaces pieced 
together from disconnected raw data files with little 
additional curatation.  The files inluded in this 
group of data sets encompass an array of file types 
and formats, including \FASTA{} (which stands for Fast-All, 
a genomics format), \SRA{} (Sequence Read Archive, for 
\DNA{} sequencing), \PDB{} (Protein Data Bank,  
representing the \ThreeD{} geometry of protein 
molecules), \MAP{} (Electron Microscopy Map), \EPS{} 
(Embedded Postscript), and \CSV{} (comma-separated values).  
There are also tables represented in Microsoft Word 
or Excel formats.  Although these various formats are 
reasonable for storing raw data, not all of them 
are actually machine-readable; in particular, 
the \EPS{}, Word, and Excel files need manual processing 
in order to use the information they provide in a 
computational manner.  A properly curated data collection 
would instead, as much as possible, unify disparate sources 
into a common machine-readable representation (such as \XML{}).}

\p{Going further, productive data curation would also 
aspire to \textit{semantic} integration, unifying disparate 
sources into a common data model.  For example, multiple 
spreadsheets among the Springer Nature Covid-19 data sets 
hold sociodemographic and epidemiological information relevant 
to modeling the spread of the disease.  These different 
sources could certainly be integrated into a canonical 
social-epidemiology-based representational paradigm which 
recognizes the disparate data points which might be 
relevant for tracking the spread of Covid-19 (with the 
potential to unify data from many countries and 
jurisdictions).  This is not only a matter ofe data
\textit{representation} (viz., how data is physically 
laid out), but also of data types and computer code.  
According to the Research Object protocol, 
data sets should include a code base 
which provides convenient computational access to the 
published data.  In the case of Covid-19, this entails 
creating a sociodemographic and epidemiological code 
library optimized for Covid-19 information, which would 
be the primary access point for researchers seeking to 
use the data which has been published in conjunction with 
the 43 manuscripts examined here that were aggregated 
on Springer Nature, along with any other coronavirus 
research which comes online.  Similar comments 
apply not only to tabular data represented in spreadsheet 
or \CSV{} form, but to more complex molecular or 
microscopy data that needs specialized scientific software 
to be properly visualized.}

\p{Considering the overall space of Covid-19 data, it is 
unavoidable that some files require special applications 
and cannot be directly merged with the overall collection.  
For instance, there is no coherent semantics for 
unifying Protein Data Bank files with social-epidemiology; 
files of the former type have specific scientific uses and 
can only be understood by special-purpose software.  
Nevertheless, a well-curated data-set collection can 
make use of such special-purpose data as convenient as 
possible.  In the case of Protein Data Bank, a downloadable 
Covid-19 archive can include source code for \IQmol{}, a 
molecular-visualization application that supports 
\PDB{} (among other file formats) and has few 
external dependencies (so it is relatively easy to 
build from source).}  

\p{Indeed, a curated Covid-19 archive 
might include an enhanced version of \IQmol{} prioritizing 
Covid-19 research, with the goal of integrating biomolecular 
and social-epidemiological data as much as possible.  
For example, as Covid-19 potentially mutates in different 
ways in different geographic areas, it will be important 
to model the connections between \q{hard} scientific 
Covid-19 information and sociodemographics.  
As the pandemic evolves, genomic and biochemical information 
may be linked to particular strains of virus, which 
in turn are linked to sociodemographic profiles: certain 
strains will be more prevalent in different populations.  
Consequently, models of Covid-19 variation will need to be 
formulated and then integrated with both chemical/molecular 
data and sociodemographic/epidemiological data.  Different 
Covid-19 strains then form a bridge linking these different 
sorts of information; researchers should be able to pass 
back and forth from molecular or genomic visualizations of 
Covid-19 to social-epidemiological charts and tables based 
on viral strains.  Ideally, capabilities for this 
sort of interdisciplinary data integration would be 
provide by a Covid-19 archive as enhancements to applications, 
such as \IQmol{}, that scientists would use to study the 
published data.}

\p{Logistically speaking, not all Covid-19 data is practical 
to reuse as a downloadable package.  This is especially 
true for genomics; several of the aforementioned 
43 coronavirus papers included data published via 
online data banks capable of hosting data sets that 
are too large for an ordinary computer.  In these 
situation scientists formulate queries or 
analytic scripts that are sent remotely to the online 
repositories, so that researchers access the actual 
published data only indirectly.  Nevertheless, access to 
these data sets can still be curated as part of 
a Covid-19 package; in particular, computer code 
can be provided which automates the process of 
networking with remote genomics archives through the 
accession numbers and file-formats which those archives 
recognize.  A case-study in this type of technology is 
provided by Verily, the Alphabet division which 
(according to news reports, such as a New York Times
cover-page story on March 15th) is developing an 
online platform for the public to check symptoms and 
locate providers prepared to treat Covid-19 cases.  
One Verily project is PurpleData, esssentially a 
miniature version of Google's \q{BigData} platform; 
unlike its larger archetype, PurpleData is designed 
to run and host data sets small enough for a single 
computer, so as to develop and test analytic queries 
which are eventually posted to remote BigData services.  
In short, PurpleData is a simulation of BigData designed 
for prototyping code which will interface with BigData.  
This prototyping/simulation paradigm is a useful model 
to apply to other remote-analytics services, such as 
the National Center for Biotechnology Information (\NCBI{}) 
genomics archives 
where some of emerging Covid-19 data has been stored.  
The point is not to recommend a direct generalization of 
PurpleData --- indeed, PurpleData, implemented in python, is 
not an ideal case-study in prototyping/simulation techniques.  
One could argue that frameworks based on WhiteDB --- 
an under-appreciated database engine (an \SQL{}/\NoSQL{} hybrid) 
currently used for 
certain CyberPhysical and 
telemedical systems --- would be more definitive prototyping 
solutions.  WhiteDB is a standalone \CCpp{} engine which can 
be dropped as source code into any compiled project; 
it therefore offers fully transparent access to data-serialization 
code, and would be especially useful for emulating the 
structural and persistence methodologies implemented 
at large scale by cloud-analytic services such as 
BigData or \NCBI{} Nucleotide/GenBank.}

\p{This discussion has highligted limitations of data 
sets published in conjunction with coronavirus articles made 
available as open-access recources on SpringerNature.  
The point here is not to criticize the work of individual 
authors, but rather to argue for a distinct data-curation 
stage in the publication process, with data curators 
playing a role distinct from that of both authors and editors.  
Moreover, the discussion has hopefully highlighted problems 
with current data-sharing paradigms, even those such 
as the Research Object and \FAIR{} initiatives which are 
explicitly devoted to improving how open-access data sets are 
published.  The Covid-19 case study documents several 
lacunae in the Research Object protocol, for example, 
which point to the need for a more detailed extension 
of this protocol.  In particular, an enhanced protocol 
should encompass: 

\begin{enumerate}

\item{} A canonical framework for archiving collections 
of data sets, not only single data sets (and not only 
groups of data sets published with a single research 
paper).  For example, all data sets published alongside 
the 43 Springer Nature articles could be unified into a 
single collection.

\item{} A code base accompanying data-set collections 
designed to help research unify the infrmation provided.  
Curating the overall collection would involve pooling 
disparate data into common representation, and 
implementing computer code which deserializes and processes 
the unified data accordingly.  For instance, \CSV{}, 
\EPS{}, and Microsoft Word/Excel tables could be migrated 
to \XML{}, \JSON{}, or 
a more complex common format (Chapter 3 of 
\textit{Advances in Ubiquitous Computing} presents the theoretical 
case for a \q{software-centric} representational format based 
on hypergraphs).  Customized computer code could then 
be implemented specifically to parse and merge the 
information present in single data sets within the 
overall collection.  This implementation would 
reciprocate the Research Object goal of unifying 
code and data, but again would operate at the level 
of an aggregate of research projects rather than a 
single Research Object.

\item{}  A unified data-set collection should provide 
prototyping and remote-access tools to interface with 
web-based information spaces that host data sets 
too large to be individually downloaded.  Ideally, 
these would include simulations of remote services 
analogous to PurpleData \visavis{} BigData, which 
would help scientists understand the design of 
the remote archives and how to interface with them.

\item{}  A unified research portal should also 
influence the design of the web portals where associated 
texts are published.  It should be easy for readers to 
identify which articles have supplemental data files and 
to download those files if desired.  Moreover, 
textual links should be established between publication 
content and data sets --- for instance, a plot or 
diagram illustrating statistical or equational distributions 
should link to the portion of the data set from which that 
quantitative data is derived.
\end{enumerate}}

\p{The above discussion has considered the Springer Nature 
articles as a case-study, but analogous comments would 
apply to other Covid-19 related resources.  For 
example, John Hopkins University has created and deployed 
a Covid-19 \q{dashboard} tracking the spread of the virus; 
new data from which the web dashboard is generated is published 
via a \GIT{} archive roughly once daily.  If and 
when the reported Verily portal comes online, hopefully 
machine-readable access to that public data will be 
provided either via an anlogous updated archive or 
via an \API{}.  Ideally, these disparate Covid-19 
projects will be interoperable: any code published 
in relation to the Springer Nature coronavirus 
collection, for example, could include components 
implemented to access the John Hopkins and 
(anticipated) Verily data sets as well as all 
the data brought in via the Springer Nature articles.  
Insofar as conscious effort is made to integrate 
all publicly accessible Covid-19 data via 
an overarching toolkit, it will be easier to continually 
accumulate new data sources as these come online.}

\p{This discussion has also used the Covid-19 crisis as a 
lense through which to examine data-publishing limitations 
in general.  These problems are not specific to 
coronavirus, but the almost unprecedented 
urgency of this epidemic exposes how science and the 
publishing industry are still struggling to 
develop technologies and practices which keep pace 
with the intersecting needs of systematic research 
and public policy.  An optimistic projection is 
that the crisis will spur momentum toward a more 
sophisticated data-sharing paradigm --- perhaps a 
generalization of the Research Object protocol 
toward data-set collections, with features as outlined 
above.  We hope to contribute to the emergence of 
such a protocol, so as to operationalize some 
of the ideas laid out in \textit{Advances in Ubiquitous 
Computing}.  It would be especially rewarding if 
an integrated data-set collection devoted to 
Covid-19 would serve as a first example and a test-bed 
for this new paradigm, given the potential public 
benefit of unifying disparate Covid-19 data 
as effectively as possible, where this technology 
can then be generalized to other medical 
priorities and other academic disciplines overall.}

%\p{}

\vspace{1.5em}
%\noindent\lun{ETS\textsc{pf} for Scientific and Technical Applications}

\end{document}


